libraries/Microsoft.Bot.Builder.Ai/TranslationMiddleware.cs

```diff
@@ -13,17 +13,54 @@
 
 namespace Microsoft.Bot.Builder.Ai
 {
-    public class TranslationMiddleware : IReceiveActivity, ISendActivity
+    public class TranslationMiddleware : IReceiveActivity
     {
-        private LuisClient luisClient;
-        private string[] nativeLanguages;
-        private Translator translator;        
+        private readonly string[] nativeLanguages;
+        private Translator translator;
+        private readonly string templatesDir;
+        private readonly Func<IBotContext, string> _getUserLanguage;
+        private readonly Func<IBotContext, Task<bool>> _setUserLanguage;
 
-        public TranslationMiddleware(HttpClient httpClient, string[] nativeLanguages, string translatorKey, string luisAppId, string luisAccessKey)
+        /// <summary>
+        /// Constructor for automatic detection of user messages
+        /// </summary>
+        /// <param name="nativeLanguages">List of languages supported by your app</param>
+        /// <param name="translatorKey"></param>
+        public TranslationMiddleware(string[] nativeLanguages, string translatorKey)
         {
             this.nativeLanguages = nativeLanguages;
-            this.luisClient = new LuisClient(luisAppId, luisAccessKey);
-            this.translator = new Translator(translatorKey, httpClient);
+            this.translator = new Translator(translatorKey);
+            templatesDir = "";
+        }
+
+        /// <summary>
+        /// Constructor for automatic of user messages and using templates
+        /// </summary>
+        /// <param name="nativeLanguages">List of languages supported by your app</param>
+        /// <param name="translatorKey"></param>
+        /// <param name="templatesDir">Directory containing no translate templates</param>
+        public TranslationMiddleware(string[] nativeLanguages, string translatorKey, string templatesDir)
+        {
+            this.nativeLanguages = nativeLanguages;
+            this.translator = new Translator(translatorKey);
+            this.templatesDir = templatesDir;
+        }
+        
+        /// <summary>
+        /// Constructor for developer defined detection of user messages and using templates
+        /// </summary>
+        /// <param name="nativeLanguages">List of languages supported by your app</param>
+        /// <param name="translatorKey"></param>
+        /// <param name="templatesDir">Directory containing no translate templates</param>
```

Comment:
You should separate out, specifying the no translate patterns which should be an in-memory data structure from using the file system.  You should not require using the file system and you should be able to manipulate the no translate patterns.

Reply:
Updated to become a new data structure stored in memory instead of accessing file system